Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnSearch.hs

```diff
@@ -193,6 +193,7 @@ buildQuote ::
 buildQuote requestId providerInfo now QuoteInfo {..} = do
   uid <- generateGUID
   tripTerms <- buildTripTerms descriptions
+  searchRequest <- QSearchReq.findById requestId >>= fromMaybeM (SearchRequestDoesNotExist requestId.getId)
```

Comment:
Instead of fetching searchRequest and then getting merchant id from it. Pass merchantId from where calling this function.